torch/masked/maskedtensor/binary.py

```diff
@@ -0,0 +1,206 @@
+    return b.get_mask()
+
+
+def _binary_helper(fn, args, kwargs, inplace):
+    from .passthrough import _map_mt_args_kwargs, _wrap_result
+
+    if len(kwargs) != 0:
+        raise ValueError("len(kwargs) must equal 0")
+    if len(args) > 2:
+        for a in args[1:]:
+            if torch.is_tensor(a):
+                raise TypeError("MaskedTensor binary ops do not support Tensor arguments")
+
+    if not _masks_match(*args[:2]):
+        raise ValueError(
+            "Input masks must match. If you need support for this, please open an issue on Github."
+        )
+
+    if (
+        is_masked_tensor(args[0])
+        and is_masked_tensor(args[1])
+        and args[0].layout() != args[1].layout()
+    ):
+        raise ValueError(
+            "Inputs should match sparsity/density. If you need support this, please open an issue on Github."
+        )
+
+    data_args, data_kwargs = _map_mt_args_kwargs(
+        args, kwargs, lambda x: x.get_data()
+    )
+    mask_args, mask_kwargs = _map_mt_args_kwargs(
+        args, kwargs, lambda x: x.get_mask()
+    )
+
+    if args[0].layout() == torch.sparse_coo:
+        if is_masked_tensor(data_args[1]):
+            if not _tensors_match(data_args[0].indices(), data_args[1].indices()):
+                raise ValueError(
+                    "sparse_coo indices must match. If you need support for this, please open an issue on Github."
+                )
+            if data_args[0].size() != data_args[1].size():
+                raise ValueError("input1 and input2 must have the same size for binary functions.")
+
+        i = data_args[0].indices()
+        size = data_args[0].size()
+        data_args[0] = data_args[0].values()
+        if is_masked_tensor(args[1]):
+            data_args[1] = data_args[1].values()
+        v = fn(*data_args)
+        result_data = torch.sparse_coo_tensor(i, v, size)
+
+    elif args[0].layout() == torch.sparse_csr:
+        if is_masked_tensor(data_args[1]) and not (
+            _tensors_match(data_args[0].crow_indices(), data_args[1].crow_indices())
+            and _tensors_match(
+                data_args[0].col_indices(), data_args[1].col_indices()
+            )
+        ):
+            raise ValueError(
+                "sparse_csr indices must match. If you need support for this, please open an issue on Github."
+            )
+
+        crow = data_args[0].crow_indices()
+        col = data_args[0].col_indices()
+        data_args[0] = data_args[0].values()
+        if is_masked_tensor(args[1]) or (
+            torch.is_tensor(args[1]) and args[1].layout == torch.sparse_coo
+        ):
+            data_args[1] = data_args[1].values()
+        v = fn(*data_args)
+        result_data = torch.sparse_csr_tensor(crow, col, v)
+
+    else:
+        result_data = fn(*data_args)
+
+    if inplace:
+        args[0]._set_data_mask(result_data, mask_args[0])
+        return args[0]
+    else:
+        result_mask = _get_at_least_one_mask(*args[:2])
```

Comment:
does this mean for non-inplace ops we're not sure if args[0] is a masked tensor?

Reply:
Yeah that's right